File: problem-2/author-solutions/s.cpp
#include<iostream>
using namespace std;
bool isPrime(long long n) {
  for (long long i = 2; i*i <= n; i++) {
    if (n % i == 0) return false;
  }
  return true;
}
int main(void) {
  long long p, q; cin >> p >> q;
  long long pq = stoll(to_string(p) + to_string(q));
  cout << ((isPrime(p) && isPrime(q) && isPrime(pq)) ? "Yes" : "No");
  return 0;
}

File: problem-2/author-solutions/s.py
def isPrime(n):
  i = 2
  while i*i <= n:
    if n % i == 0:
      return False
    i += 1
  return True

p, q = [int(x) for x in input().split()]
pq = int(str(p) + str(q))
print("Yes" if (isPrime(p) and isPrime(q) and isPrime(pq)) else "No")